tgui/packages/tgui/interfaces/KitchenMachine.js

```diff
@@ -0,0 +1,95 @@
+import { useBackend } from '../backend';
+import { Button, Section, Table, Flex, Icon, Dimmer } from '../components';
+import { Window } from '../layouts';
+
+export const KitchenMachine = (props, context) => {
+  const { data } = useBackend(context);
+  const { ingredients } = data;
+
+  return (
+    <Window resizable>
+      <Window.Content scrollable display="flex" className="Layout__content--flexColumn">
+        <Operating/>
+        <KitchenTop/>
+        <Section title="Ingredients">
+          <Table>
+            {ingredients.map((product) => (
+              <Table.Row key={product.name}>
+                <Table.Cell bold>{product.name}</Table.Cell>
+                <Table.Cell collapsing textAlign="center">
+                  {product.amount} {product.units}
+                </Table.Cell>
+              </Table.Row>
+            ))}
+          </Table>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const KitchenTop = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { inactive,
+          tooltip
+        } = data;
+
+  return (
+    <Section title="Controls">
+      <Flex>
+        <Flex.Item width="50%" mr="3px">
+          <Button
+            fluid
+            textAlign="center"
+            icon="power-off"
+            disabled={inactive}
+            tooltip={tooltip}
+            tooltipPosition="bottom"
+            content="Activate"
+            onClick={() => act('cook')}
+          />
+        </Flex.Item>
+        <Flex.Item width="50%" mr="3px">
+          <Button
+            fluid
+            textAlign="center"
+            icon="eject"
+            disabled={inactive}
+            tooltip={inactive ? tooltip : ""}
+            tooltipPosition="bottom"
+            content="Eject Contents"
+            onClick={() => act('eject')}
+          />
+        </Flex.Item>
+      </Flex>
+    </Section>
+  );
+}
+
+const Operating = (props, context) => {
+  const { data } = useBackend(context);
+  const { operating,
```

Comment:
could you make the data const here and in other blocks be consistent with what I suggested previously